.github/workflows/ci.yml

```
@@ -105,3 +105,19 @@ jobs:
         sudo ./script/build-libgit2.sh --static --system
     - name: Test
       run: go test --count=1 --tags "static,system_libgit2" ./...
+
+  check-generate:
+    name: Check generated files were not modified
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: '1.17'
+      id: go
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+    - name: Generate files
+      run: make generate
+    - name: Check nothing changed
+      run: git diff --quiet --exit-code || (echo "detected changes after generate" && exit 1)
```

```suggestion
      run: git diff --quiet --exit-code || (echo "detected changes after generate" ; git status ; exit 1)
```

this way the caller will be able to see what were the changes.